skbio/stats/distance/_permdisp.py

```diff
@@ -0,0 +1,269 @@
+    ...                    's5': 'G2', 's6': 'G2'}})
+    >>> # make output deterministic; not necessary for normal use
+    >>> np.random.seed(0)
+    >>> permdisp(dm, df, 'Grouping', permutations=99, test='centroid')
+    method name               PERMDISP
+    test statistic name        F-value
+    sample size                      6
+    number of groups                 2
+    test statistic             3.67082
+    p-value                       0.29
+    number of permutations          99
+    Name: PERMDISP results, dtype: object
+
+    Note that when providing a ``DataFrame``, the ordering of rows and/or
+    columns does not affect the grouping vector that is extracted. The
+    ``DataFrame`` must be indexed by the distance matrix IDs (i.e., the row
+    labels must be distance matrix IDs).
+
+    If IDs (rows) are present in the ``DataFrame`` but not in the distance
+    matrix, they are ignored. The previous example's ``s7`` ID illustrates this
+    behavior: note that even though the ``DataFrame`` had 7 objects, only 6
+    were used in the test (see the "Sample size" row in the results above to
+    confirm this). Thus, the ``DataFrame`` can be a superset of the distance
+    matrix IDs. Note that the reverse is not true: IDs in the distance matrix
+    *must* be present in the ``DataFrame`` or an error will be raised.
+
+    PERMDISP should be used to determine wether the dispersions between the
+    groups in your distance matrix are significantly separated.
+    A non-significant test result indicates that group dispersions are similar
+    to eachother. PERMANOVA or ANOSIM should then be used in conjunction to
+    determine wether clustering within groups is significant.
+
+    """
+
+    ordination = pcoa(distance_matrix)
+    samples = ordination.samples
+
+    sample_size, num_groups, grouping, tri_idxs, distances = _preprocess_input(
+        distance_matrix, grouping, column)
+
+    if test == 'centroid':
+        test_stat_function = partial(_compute_centroid_groups, samples)
+    elif test == 'median':
+        test_stat_function = partial(_compute_median_groups, samples)
+    else:
+        raise ValueError('Test must be centroid or median')
+
+    stat, p_value = _run_monte_carlo_stats(test_stat_function, grouping,
+                                           permutations)
+
+    return _build_results('PERMDISP', 'F-value', sample_size, num_groups,
+                          stat, p_value, permutations)
+
+
+def _compute_centroid_groups(samples, grouping):
+
+    groups = []
+
+    samples['grouping'] = grouping
+
+    centroids = samples.groupby('grouping').aggregate('mean')
+
+    for label, df in samples.groupby('grouping'):
+        groups.append(cdist(df.values[:, :-1], [centroids.loc[label].values],
+                      metric='euclidean'))
+
+    stat, _ = f_oneway(*groups)
+    stat = stat[0]
+
+    return stat
+
+
+def _config_med(x):  # not explicitly tested by permdisp test suite
```

Comment:
We should have some basic tests, in part to know that we'll catch a change in the behavior of `hdmedians` if that ever happens.

Reply:
will do!

Reply:
should be done now ✅